### ISSUE TITLE
Sometimes runtime error occurs

### ISSUE THREAD

**Issue by kkr881** [Zibbp/ganymede#257]
I am using Ganymede with Docker on my Synology NAS.

Sometimes Ganymede-api throws a runtime error and stops some queues.
When Ganymede-api crashes, Docker restarts it, so Ganymede-api will restart, but queues created before the restart will become dead queues, preventing new queue creation.
Is it possible to solve the above problem?

The log at the time of the error is as follows
```
panic: runtime error: index out of range [0] with length 0
goroutine 2443 [running]:
github.com/zibbp/ganymede/internal/archive.(*Service).TaskVodSaveLiveInfo(0xc0005980f0, 0xc0007edd00, 0xc000851860, 0xc0004dc9a0, 0x1)
	/app/internal/archive/archive.go:605 +0xa69
created by github.com/zibbp/ganymede/internal/archive.(*Service).TaskVodDownloadLiveThumbnail
	/app/internal/archive/archive.go:504 +0x56d

```
Attached is the Docker log
[_ganymede-api_logs.txt](https://github.com/Zibbp/ganymede/files/12113485/_ganymede-api_logs.txt)

**Comment by Zibbp** [comment]
Can you enable debug logging in the mounted data dir, `/data/config.json` and post logs again when this happens? Do you notice a specific time this happens? Randomly?

**Comment by kkr881** [comment]
It's happens Randomly..
I don't know the conditions that arise

**Comment by kkr881** [comment]
@Zibbp 
Hello.
After the phenomenon occurred, we checked the reproduction of debug as enable.
The error is different from the previous one, but I got a log with debug enabled, so I'm sending you a log with the error and error.

This error occurred on 2023/08/10 05:12:14.
You can refer to it in the corresponding time zone of the log.
```
panic: ent: queue not found

goroutine 72391 [running]:
github.com/zibbp/ganymede/ent.(*QueueUpdateOne).SaveX(...)
/app/ent/queue_update.go:978
github.com/zibbp/ganymede/internal/archive.(*Service).TaskVodDownloadLiveThumbnail(0xc0000a1f80, 0xc0000be300, 0xc000192780, 0xc0001689a0, 0x0)
/app/internal/archive/archive.go:486 +0x1034
created by github.com/zibbp/ganymede/internal/archive.(*Service).RefreshLiveThumbnails
/app/internal/archive/archive.go:546 +0x118
```

Attached is the Docker log
[ganymede-api.txt](https://github.com/Zibbp/ganymede/files/12322600/ganymede-api.txt)

**Comment by Zibbp** [comment]
Thanks for the report. It looks like the queue item is gone when attempting to update the livestream thumbnail. I've pushed a commit which checks if the queue item exists and return if it doesn't. This should prevent the application from crashing and instead log a warning. https://github.com/Zibbp/ganymede/pull/269

**Comment by Zibbp** [comment]
Fix for this error has been released in [v1.4.1](https://github.com/Zibbp/ganymede/releases/tag/v1.4.1)

**Comment by russelg** [comment]
Hi Zibbp,

I've upgraded to v1.4.1, but that hasn't resolved the issue that this ticket was originally for.

I finally have the debug logs for the failure I've experienced:

```
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:28+08:00","message":"getting live streams using the following query param: ?user_login=shutupandjam_ne&user_login=polytoneofficial&user_login=jenkinsfatty&user_login=ani
sonhijack&user_login=mogra&user_login=420jenkins&user_login=newsham&user_login=midnightsnacksfm&user_login=otographic"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:32+08:00","message":"finished downloading live video for 49104708813"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:32+08:00","message":"streamlink detected the stream was down - killing chat_downloader"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:32+08:00","message":"getting video duration"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:32+08:00","message":"video duration: 8061"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:32+08:00","message":"starting task video convert for vod 83c8d3df-3be8-11ee-8f5d-0242ac120006"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:32+08:00","message":"video convert args: [-y -hide_banner -i /tmp/49104708813_83c8d3df-3be8-11ee-8f5d-0242ac120006-video.mp4 -c:v copy -c:a copy /tmp/49104708813_83c8d3d
f-3be8-11ee-8f5d-0242ac120006-video-convert.mp4]"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:33+08:00","message":"finished downloading live chat for 49104708813"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:33+08:00","message":"moving file: /tmp/49104708813_83c8d3df-3be8-11ee-8f5d-0242ac120006-live-chat.json to /vods/midnightsnacksfm/2023-08-16-49104708813-live-83c8d3df-3be
8-11ee-8f5d-0242ac120006/49104708813-live-chat.json"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:34+08:00","message":"running check live stream schedule"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:34+08:00","message":"checking live channels"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:34+08:00","message":"getting live streams using the following query param: ?user_login=shutupandjam_ne&user_login=polytoneofficial&user_login=jenkinsfatty&user_login=ani
sonhijack&user_login=mogra&user_login=420jenkins&user_login=newsham&user_login=midnightsnacksfm&user_login=otographic"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:34+08:00","message":"midnightsnacksfm is now live"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:34+08:00","message":"Live notification is disabled"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:34+08:00","message":"starting task vod create folder for vod 5504759d-3bfb-11ee-8f5d-0242ac120006"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:34+08:00","message":"creating folder: midnightsnacksfm/2023-08-16-49104708813-live-5504759d-3bfb-11ee-8f5d-0242ac120006"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:36+08:00","message":"starting task vod download thumbnail for live stream 5504759d-3bfb-11ee-8f5d-0242ac120006"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:36+08:00","message":"getting live streams using the following query param: ?user_login=midnightsnacksfm"}
ganymede-api  | {"level":"error","time":"2023-08-16T14:08:36+08:00","message":"no stream found"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:36+08:00","message":"starting task chat convert for vod 83c8d3df-3be8-11ee-8f5d-0242ac120006"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:36+08:00","message":"starting task vod save info for vod 5504759d-3bfb-11ee-8f5d-0242ac120006"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:36+08:00","message":"Error notification is disabled"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:36+08:00","message":"refresh live thumbnails called...sleeping for 20 minutes"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:36+08:00","message":"getting live streams using the following query param: ?user_login=midnightsnacksfm"}
ganymede-api  | {"level":"debug","time":"2023-08-16T14:08:36+08:00","message":"getting user by login: midnightsnacksfm"}
ganymede-api  | panic: runtime error: index out of range [0] with length 0
ganymede-api  | 
ganymede-api  | 
ganymede-api  | goroutine 77738 [running]:
ganymede-api  | github.com/zibbp/ganymede/internal/archive.(*Service).TaskVodSaveLiveInfo(0xc00041c870, 0xc0000a1700, 0xc0003712c0, 0xc00055da20, 0x1)
ganymede-api  |         /app/internal/archive/archive.go:613 +0xab2
ganymede-api  | created by github.com/zibbp/ganymede/internal/archive.(*Service).TaskVodDownloadLiveThumbnail
ganymede-api  |         /app/internal/archive/archive.go:512 +0x6cd
```
The flow here is, the original midnightsnacksfm stream ended, and that was all well and good. Then the live check happens, and it returns that midnightsnacksfm is still live (for some reason). It then tries to fetch this stream, and then it falls over (you can see "no stream found").

**Comment by kkr881** [comment]
@Zibbp 
Hello

The same error as the above user occurred.
The log is as follows, and the part where the actual problem occurs seems to be the same as the user above.
my check interval is 90s.

```
29T04:39:50+09:00 DBG getting live streams using the following query param: ?user_login=sweet_lena0 
panic: runtime error: index out of range [0] with length 0
goroutine 18516 [running]:
github.com/zibbp/ganymede/internal/archive.(*Service).TaskVodSaveLiveInfo(0xc0000a28d0, 0xc000365b00, 0xc0007ac5a0, 0xc0003d89a0, 0x1)
	/app/internal/archive/archive.go:613 +0xab2
created by github.com/zibbp/ganymede/internal/archive.(*Service).TaskVodDownloadLiveThumbnail
	/app/internal/archive/archive.go:546 +0xf9f
usermod: no changes
-------------------------------------
User uid:    1026
User gid:    100
-------------------------------------
Version    : 
2023-09-29T04:40:17+09:00 2023-09-29T04:40:17+09:00 INF 2023-09-29T04:40:17+09:00 2023-09-29T04:40:17+09:00 INF 2023-09-29T04:40:17+09:00 2023-09-29T04:40:17+09:00 INF 2023-09-29T04:40:17+09:00 2023-09-29T04:40:17+09:00 INF config file found at /data/config.json, loading 
Git Hash   : d2941111ae2d3f6c8dbd7572b1bfd958736fed46
Build Time : 2023-09-10T18:34:51Z
2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG config file loaded: /data/config.json 
2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 INF 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 INF 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 INF 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 INF debug mode enabled 
2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG setting up database connection 
2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG creating new handler 
2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG mapping routes 
2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG setting up twitch auth schedule 
2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG running twitch auth schedule 
2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG 2023-09-29T04:40:20+09:00 2023-09-29T04:40:20+09:00 DBG authenticating with twitch 
   ____    __
  / __/___/ /  ___
 / _// __/ _ \/ _ \
/___/\__/_//_/\___/ v4.11.1
High performance, minimalist Go web framework
https://echo.labstack.com
____________________________________O/_______
                                    O\
⇨ http server started on [::]:4000
2023-09-29T04:40:21+09:00 2023-09-29T04:40:21+09:00 INF 2023-09-29T04:40:21+09:00 2023-09-29T04:40:21+09:00 INF 2023-09-29T04:40:21+09:00 2023-09-29T04:40:21+09:00 INF 2023-09-29T04:40:21+09:00 2023-09-29T04:40:21+09:00 INF authenticated with twitch 
2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG setting up twitch categories schedule 
2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG setting up queue item schedule 
2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG setting up prune video schedule 
2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG running set twitch categories schedule 
2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG running queue item schedule 
2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 INF 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 INF 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 INF 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 INF setting up check watched channel videos schedule 
2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG setting up check live stream schedule 
2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG setting video check interval to run every 180 minutes 
2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG setting live check interval to run every 90 seconds 
2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 INF 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 INF 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 INF 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 INF running check watched channel videos schedule 
2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG running check live stream schedule 
2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG checking live channels 
2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 INF 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 INF 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 INF 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 INF Checking 1 channels for new videos 
2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG 2023-09-29T04:40:25+09:00 2023-09-29T04:40:25+09:00 DBG no queue items are on hold 
```

**Comment by Zibbp** [comment]
I believe what is happening is the schedule live channel check sees an active livestream returned by the Twitch API. Then a few seconds later when the task to save the JSON info of the livestream, the Twitch API finally updates and the stream goes offline.

I could catch the error where the JSON is saved for the archive but that would allow the archive to continue and eventually fail out when there is no thumbnail or video to download. I think the better course of action is to catch the error [where it's been failing](https://github.com/Zibbp/ganymede/blob/view-count/internal/archive/archive.go#L613) and simply delete the queue item if it fails there. If the endpoint failed to return the JSON of the stream, it's probably safe to assume the stream has ended.

I'm working on the fixes in https://github.com/Zibbp/ganymede/pull/301

**Comment by Zibbp** [comment]
Released in [v1.4.3](https://github.com/Zibbp/ganymede/releases/tag/v1.4.3).
